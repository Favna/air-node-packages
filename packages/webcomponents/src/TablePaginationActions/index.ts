export { default as TablePaginationActions, TablePaginationActionsProps } from './TablePaginationActions';