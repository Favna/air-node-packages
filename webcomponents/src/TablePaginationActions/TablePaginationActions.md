### Usage of the TablePaginationActions component:

**Examples:**

```tsx
import React from 'react'; 
import TablePagination from '@material-ui/core/TablePagination'; 
import { TablePaginationActions } from '@rws-air/webcomponents'; 

  <TablePagination
      labelRowsPerPage={'rows per page'}
      labelDisplayedRows={({from, to, count}) => `${from <= 9 ? ` 0${from} ` : from}-${to} of ${count}` }
      component='div'
      count={20}
      onChangePage={() => console.log('Changed Page!')}
      onChangeRowsPerPage={() => console.log('Changed amount of rows per page')}
      page={0}
      rowsPerPage={5}
      rowsPerPageOptions={[5, 10, 15, 20]}
      ActionsComponent={TablePaginationActions}
    />
```

```jsx
import React from 'react'; 
import TablePagination from '@material-ui/core/TablePagination'; 
import { TablePaginationActions } from '@rws-air/webcomponents'; 

  <TablePagination
      labelRowsPerPage={'rows per page'}
      labelDisplayedRows={({from, to, count}) => `${from <= 9 ? ` 0${from} ` : from}-${to} of ${count}` }
      component='div'
      count={20}
      onChangePage={() => console.log('Changed Page!')}
      onChangeRowsPerPage={() => console.log('Changed amount of rows per page')}
      page={0}
      rowsPerPage={5}
      rowsPerPageOptions={[5, 10, 15, 20]}
      ActionsComponent={TablePaginationActions}
    />
```

