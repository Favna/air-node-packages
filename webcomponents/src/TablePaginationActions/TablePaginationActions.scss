@import '../styles/vars.scss';

.paginationNumbers {
  color: $blue;
  cursor: pointer;
  font-family: $rws-font-text;
  font-weight: bolder;
  margin-left: 0.5vw;
  margin-right: 0.5vw;
}

.activePageChip {
  border-width: 2px !important;
  font-weight: bolder;
}

.noHoverBackground:hover,
.noHoverBackground:focus {
  background-color: transparent !important;
}

.ie11-chipCorrection {
  min-width: 2rem;
}
