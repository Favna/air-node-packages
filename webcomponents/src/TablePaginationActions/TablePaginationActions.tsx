import React, { FC, SyntheticEvent, Fragment } from "react";
import classnames from 'classnames';
import css from './TableBodyCell.scss';
import Chip from '@material-ui/core/Chip';
import IconButton from '@material-ui/core/IconButton';
import FirstPageIcon from '@material-ui/icons/FirstPage';
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';
import LastPageIcon from '@material-ui/icons/LastPage';

export type TablePaginationActionsProps = {
    rowsPerPage: number;
    page: number;
    count: number;
    onChangePage: (event: any, page: number) => void;
}

const TablePaginationActions: FC<TablePaginationActionsProps> = props => {
    const handleFirstPageButtonclick = (event: any) => props.onChangePage(event, 0);
    const handleBackButtonClick = (event: any) => props.onChangePage(event, props.page - 1);
    const handleNextButtonClick = (event: any) => props.onChangePage(event, props.page + 1);
    const handleLastPageButtonClick = (event: any) => props.onChangePage(event, Math.max(0, Math.ceil(props.count / props.rowsPerPage) - 1));
    const handlePageClick = (page: number) => (event: SyntheticEvent) => props.onChangePage(event, page);

    const renderPages = () => (
        [...Array(Math.ceil(props.count / props.rowsPerPage)).keys()]
            .map(x => ++x)
            .map((page, key) => {
                if (props.page + 1 === page) {
                    return <Chip key={key} label={page} color='primary' variant='default' onClick={handlePageClick(page - 1)} className={classnames(css.activePageChip, css['ie11-chipCorrection'])} />
                }

                return <span key={key} onClick={handlePageClick(page - 1)} className={classnames(css.paginationNumbers)}>{page}</span>
            })
    )

    return (
        <Fragment>
            <IconButton onClick={handleFirstPageButtonclick} disabled={props.page === 0} color='primary' className={css.noHoverBackground}>
                <FirstPageIcon />
            </IconButton>
            <IconButton onClick={handleBackButtonClick} disabled={props.page === 0} color='primary' className={css.noHoverBackground}>
                <KeyboardArrowLeft />
            </IconButton>
            {renderPages()}
            <IconButton onClick={handleNextButtonClick} disabled={props.page >= Math.ceil(props.count / props.rowsPerPage) - 1} color='primary' className={css.noHoverBackground}>
                <KeyboardArrowRight />
            </IconButton>
            <IconButton onClick={handleLastPageButtonClick} disabled={props.page >= Math.ceil(props.count / props.rowsPerPage) - 1} color='primary' className={css.noHoverBackground}>
                <LastPageIcon />
            </IconButton>
        </Fragment>
    )
}

export default TablePaginationActions;