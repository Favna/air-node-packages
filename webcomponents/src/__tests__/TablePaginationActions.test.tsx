import { IconButton } from '@material-ui/core';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import TablePaginationActions from '../TablePaginationActions/TablePaginationActions';

let actions: ShallowWrapper;
const onChangePage = jest.fn();

beforeAll(() => actions = shallow(
  <TablePaginationActions rowsPerPage={10} page={0} count={42}
    onChangePage={onChangePage} />
));

test('should match snapshot', () => {
  expect(actions).toMatchSnapshot();
});

test('should jump to first page', () => {
  const goToFirstPageButton = actions.find(IconButton).first();

  goToFirstPageButton.simulate('click');
  expect(onChangePage).toHaveBeenCalled();
});

test('should jump to last page', () => {
  const goToLastPageButton = actions.find(IconButton).last();

  goToLastPageButton.simulate('click');
  expect(onChangePage).toHaveBeenCalled();
});

test('should jump to next page', () => {
  const goToNextPageButton = actions.find(IconButton).at(2);

  goToNextPageButton.simulate('click');
  expect(onChangePage).toHaveBeenCalled();
});

test('should jump to previous page', () => {
  const goToPreviousPageButton = actions.find(IconButton).at(1);

  goToPreviousPageButton.simulate('click');
  expect(onChangePage).toHaveBeenCalled();
});