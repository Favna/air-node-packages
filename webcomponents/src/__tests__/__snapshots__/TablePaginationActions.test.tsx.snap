// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Styled(MuiBox)
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <WithStyles(ForwardRef(IconButton))
    color="primary"
    disabled={true}
    onClick={[Function]}
  >
    <FirstPageIcon />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(IconButton))
    color="primary"
    disabled={true}
    onClick={[Function]}
  >
    <KeyboardArrowLeftIcon />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(Chip))
    className=""
    color="primary"
    key="0"
    label={1}
    onClick={[Function]}
    variant="default"
  />
  <span
    className=""
    key="1"
    onClick={[Function]}
  >
    2
  </span>
  <span
    className=""
    key="2"
    onClick={[Function]}
  >
    3
  </span>
  <span
    className=""
    key="3"
    onClick={[Function]}
  >
    4
  </span>
  <span
    className=""
    key="4"
    onClick={[Function]}
  >
    5
  </span>
  <WithStyles(ForwardRef(IconButton))
    color="primary"
    disabled={false}
    onClick={[Function]}
  >
    <KeyboardArrowRightIcon />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(IconButton))
    color="primary"
    disabled={false}
    onClick={[Function]}
  >
    <LastPageIcon />
  </WithStyles(ForwardRef(IconButton))>
</Styled(MuiBox)>
`;
